packages/paste-website/src/components/tokens-list/index.tsx

```diff
@@ -1,136 +1,88 @@
-  }, [filterString, props, theme]);
+    if (filterString === '') {
+      setNoResults(0);
+    }
+  }, [filterString]);
 
-  const handleInput = (e: React.FormEvent<HTMLInputElement>): void => {
-    const filter = e.currentTarget.value;
-    setFilterString(filter);
-  };
+  // The rendered tokens should update every time the filterString, props, or theme changes
+  // @TODO add tracking back: trackTokenFilterString(filterString)
 
-  const uid = useUID();
+  const onChangeCallback: React.ChangeEventHandler<HTMLInputElement> = React.useCallback(({target: {value}}) => {
+    setFilterString(value);
+  }, []);
+  const handleOnChange = React.useMemo(() => debounce(onChangeCallback, 150), []);
+  const handleOnClearSearch: VoidFunction = () => {
+    setFilterString('');
+    setNoResults(0);
+  };
 
-  if (Tokens === null) {
-    return <NoTokensFound onClearSearch={() => setFilterString('')} />;
+  // @TODO make compatible with the category section layout structure.
+  if (showNullState) {
+    return <NoTokensFound onClearSearch={handleOnClearSearch} />;
   }
 
   return (
     <ContentWrapper>
-      <PageAside
-        data={{
-          fileAbsolutePath: '',
-          frontmatter: {slug: '/tokens/list', title: 'Design tokens'},
-          headings: TokenKeys.map((value) => ({value, depth: 2})),
-        }}
-      />
+      <PageAside data={pageAsideData} />
       <Content>
         <Box paddingBottom="space80">
-          <Label htmlFor="test" id="test-label">
+          <Label htmlFor={seed('filter-string')} id={seed('filter-string-label')}>
             Filter tokens
           </Label>
           <Input
             type="text"
-            id="test"
-            aria-labelledby="test-label"
-            onChange={handleInput}
-            insertBefore={<FilterIcon decorative={false} title="Description of icon" />}
+            id={seed('filter-string')}
+            aria-labelledby={seed('filter-string-label')}
+            onChange={handleOnChange}
+            insertBefore={<FilterIcon decorative={false} title="TODO" />}
           />
         </Box>
 
-        {TokenKeys.map((tokenKey: string) => (
-          <React.Fragment key={`catname-${tokenKey}`}>
-            <AnchoredHeading as="h2" variant="heading20">
-              {sentenceCase(tokenKey)}
-            </AnchoredHeading>
-            <Box marginBottom="space160" data-cy="tokens-table-container">
-              {Tokens[tokenKey].map(({name, value, comment}: {name: string; value: string; comment: string}) => {
-                console.log('key', tokenKey);
-                return (
-                  <TokenCard key={`token${name}`} category={tokenKey} name={name} value={value} comment={comment} />
-                );
-              })}
-            </Box>
-          </React.Fragment>
+        {TOKEN_CATEGORIES.map((categoryKey) => (
+          <CategorySection
+            categoryKey={categoryKey}
+            value={filterString}
+            theme={theme}
```

Comment:
AI: might be helpful to rename this as `themeKey` to clarify that we are not passing the entire theme object here.